library/alloc/tests/slice.rs

```diff
@@ -1898,3 +1898,49 @@ fn subslice_patterns() {
     m!(&mut v, [..] => ());
     m!(&mut v, [x, .., y] => c!((x, y), (&mut N, &mut N), (&mut N(0), &mut N(4))));
 }
+
+#[test]
+fn test_group_by() {
```

Comment:
It would be nice to have elements that are equal to previous elements but are disjoint from their group. For example:

```rust
let slice = &[1, 1, 1, 3, 3, 2, 2, 2, 1];
```